src/memfd.rs

```diff
@@ -192,21 +198,44 @@ impl Memfd {
     /// Return the current sealing bitflags.
     fn file_get_seals(fp: &fs::File) -> Result<u64, crate::Error> {
         let fd = fp.as_raw_fd();
-        // UNSAFE(lucab): required syscall.
-        let r = unsafe { libc::syscall(libc::SYS_fcntl, fd, libc::F_GET_SEALS) };
+        // SAFETY: The syscall called has no soundness implications (i.e. does not mess with
+        // process memory in weird ways, checks its arguments for correctness, etc.). Furthermore
+        // due to invariants of `Memfd` this syscall is provided a valid file descriptor.
+        let r = unsafe {
+            libc::syscall(libc::SYS_fcntl, fd, libc::F_GET_SEALS)
+        };
         if r < 0 {
             return Err(crate::Error::GetSeals(std::io::Error::last_os_error()));
         };
-
         Ok(r as u64)
     }
+}
+
+impl AsRawFd for Memfd {
+    fn as_raw_fd(&self) -> RawFd {
+        self.as_file().as_raw_fd()
+    }
+}
 
-    /// Assemble a `File` object from a raw file-descriptor.
+impl IntoRawFd for Memfd {
+    fn into_raw_fd(self) -> RawFd {
+        self.into_file().into_raw_fd()
+    }
+}
+
+impl FromRawFd for Memfd {
```

Comment:
I don’t think we can write a safe conversion from `fd` function _even_ if we check that the file descriptor is a `memfd`. In particular users still would be responsible for ensuring that they don’t call `checked_from_fd` multiple times with a copy of the same fd.

If we’re adding any conversion function from a `RawFd`, I don’t see a reason to also delegate other invariants to the caller. Especially, because people _will_ eventually come up with a way to use `Memfd` in a way such that checking whether its an actual `memfd` file descriptor is a prohibitive overhead.